plugins/woocommerce-admin/client/core-profiler/pages/Loader.tsx

```diff
@@ -0,0 +1,70 @@
+/**
+ * External dependencies
+ */
+import classNames from 'classnames';
+import { useState, useEffect } from '@wordpress/element';
+/**
+ * Internal dependencies
+ */
+import { CoreProfilerStateMachineContext } from '..';
+import ProgressBar from '../components/progress-bar/progress-bar';
+import { getLoaderStageMeta } from '../get-loader-stage-meta';
+
+export type Stage = {
+	title: string;
+	image?: string | JSX.Element;
+	paragraphs: Array< {
+		label: string;
+		text: string;
+		duration?: number;
+	} >;
+};
+
+export type Stages = Array< Stage >;
+
+export const Loader = ( {
+	context,
+}: {
+	context: CoreProfilerStateMachineContext;
+} ) => {
+	const stages = getLoaderStageMeta( context.loader.useStages ?? 'default' );
+	const currentStage = stages[ context.loader.stageIndex ?? 0 ];
+	const [ currentParagraph, setCurrentParagraph ] = useState( 0 );
+
+	useEffect( () => {
+		const interval = setInterval( () => {
+			currentStage.paragraphs[ currentParagraph + 1 ]
+				? setCurrentParagraph(
+						( currentParagraph ) => currentParagraph + 1
+				  )
+				: setCurrentParagraph( 0 );
+		}, currentStage.paragraphs[ currentParagraph ]?.duration ?? 3000 );
+
+		return () => clearInterval( interval );
+	}, [ currentParagraph, currentStage.paragraphs ] );
+
+	return (
+		<div
+			className={ classNames(
+				'woocommerce-profiler-loader',
+				context.loader.className
+			) }
+		>
+			{ currentStage.image && currentStage.image }
+
+			<h1 className="woocommerce-profiler-loader__title">
+				{ currentStage.title }
+			</h1>
+			<ProgressBar
+				className={ 'progress-bar' }
+				percent={ context.loader.progress ?? 0 }
+				color={ 'var(--wp-admin-theme-color)' }
+				bgcolor={ '#E0E0E0' }
+			/>
+			<p className="woocommerce-profiler-loader__paragraph">
```

Comment:
Should we add some CSS styling to this?

Reply:
The paragraph matches the design, in my opinion. Do you think we need additional CSS?

Reply:
It seems different in my local. 🤔 

I think we need to add these styles. Also maybe we should set a `max-width`.

```
font-size: 16px;
line-height: 24px;
/* Gutenberg/Gray 800 */
color: #2F2F2F;
opacity: 0.8;
```

Actual
![Screenshot 2023-05-10 at 11 12 31](https://github.com/woocommerce/woocommerce/assets/4344253/b9c3c30f-a849-402e-ac3f-176548c716cb)

Expected
<img width="1440" alt="Loader 21" src="https://github.com/woocommerce/woocommerce/assets/4344253/01e9ba58-f30b-44fb-bca6-b3ebde17edd8">

Reply:
Sorry for missing this change 🙏 

Updated in https://github.com/woocommerce/woocommerce/pull/38019/commits/fd6e78fa0acf235e548cbe3518516857a63c8d19